pkgs/by-name/do/dorion/package.nix

```diff
@@ -1,65 +1,186 @@
+      inherit pname version src;
+      hash = "sha256-qrB+IpuAhoiKfwZvfF8RSZbiyASGIABmYUd70iqSPyw=";
+    };
+
+    nativeBuildInputs = [
+      pnpm.configHook
+      nodejs
+      copyDesktopItems
+    ];
+
+    buildPhase = ''
+      runHook preBuild
+
+      cp ${shelter} src-tauri/injection/shelter.js
+      pnpm build
+
+      runHook postBuild
+    '';
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out
+      cp -r src-tauri/{injection,html} $out
+
+      runHook postInstall
+    '';
+  };
+
+in
+
+rustPlatform.buildRustPackage {
+  inherit pname version src;
+
+  sourceRoot = "${src.name}/src-tauri";
+
+  useFetchCargoVendor = true;
+  cargoHash = "sha256-dXWw/lro8JHZsLeYO8DvqJaH8JNQh7zQ0vKjEpG8fOE=";
 
   nativeBuildInputs = [
+    pkg-config
+    yq-go
     autoPatchelfHook
-    dpkg
     wrapGAppsHook3
   ];
 
   buildInputs = [
-    glib-networking
+    openssl
+    webkitgtk_4_1
     gst_all_1.gst-plugins-bad
     gst_all_1.gst-plugins-base
     gst_all_1.gst-plugins-good
-    webkitgtk_4_0
+    glib-networking
   ];
 
-  installPhase = ''
-    runHook preInstall
+  runtimeDependencies = [
+    libappindicator
+    libayatana-appindicator
+  ];
 
-    mkdir -pv $out
-    mv -v {bin,lib,share} $out
+  env = {
+    TAURI_RESOURCE_DIR = "${placeholder "out"}/lib";
+  };
 
-    runHook postInstall
+  postPatch = ''
+    # disable pre-build script, change distribution dir from an URL to src, and disable auto-updater
+    yq -iPo=json '
+        .build.beforeBuildCommand = "" |
+        .build.frontendDist = "src" |
+        .bundle.resources = (.bundle.resources | map(select(. != "updater*")))
+    ' tauri.conf.json
+
+    (cd $cargoDepsCopy/tauri-utils-* && patch -p3 <${./tauri-env-resource-dir.patch})
+  '';
+
+  preConfigure = ''
+    cp -r ${frontend}/* ./
+  '';
+
+  postInstall = ''
+    mkdir -p $out/lib/dorion
+    cp -r {injection,html} $out/lib/dorion
+    install -Dm644 "$out/share/icons/hicolor/32x32/apps/dorion.png" "./${src.name}/src-tauri/icons/32x32icon.png"
+    install -Dm644 "$out/share/icons/hicolor/128x128/apps/dorion.png" "./${src.name}/src-tauri/icons/128x128.png"
+    install -Dm644 "$out/share/icons/hicolor/256x256/apps/dorion.png" "./${src.name}/src-tauri/icons/128x128@2.png"
+    install -Dm644 "$out/share/icons/hicolor/512x512/apps/dorion.png" "./${src.name}/src-tauri/icons/icon.png"
   '';
 
+  desktopItems = [
+    (makeDesktopItem {
+      name = "dorion";
+      genericName = "Internet Messenger";
+      desktopName = "Dorion";
+      exec = "dorion %U";
+      tryExec = "dorion %U";
+      icon = "dorion";
+      comment = "Tiny alternative Discord client";
+      keywords = [
+        "discord"
+        "vencord"
+        "tauri"
+        "chat"
+      ];
+      categories = [
+        "Network"
+        "InstantMessaging"
+        "Chat"
+      ];
+      mimeTypes = [ "x-scheme-handler/discord" ];
+      startupWMClass = "Dorion";
+      terminal = false;
+    })
+  ];
```

Comment:
ditto -- though, you may still want to final one as Tauri's generated .desktop files may be lacking in metainfo. See https://github.com/NixOS/nixpkgs/blob/cd43d6867192ddfb98ba106065a89790e2722b3f/pkgs/by-name/mo/modrinth-app-unwrapped/package.nix#L75-L89 for example